### PR TITLE
Fluent API for executing queries

### PR DESCRIPTION
**Pull request by RichardIrons-neo4j**
Everything is in experimental, and only a basic execution of the query, returning an EagerResult, is currently supported.

### CODE REVIEW

File: Neo4j.Driver/Neo4j.Driver/Experimental/FluentQueries/ExecutableQuery.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        return this;
+    }
+
+    public IExecutableQuery WithParameters(Dictionary<string, object> parameters)
+    {
+        _query = new Query(_query.Text, parameters);
+        return this;
+    }
+
+    public IExecutableQuery WithParameter(string name, object value)
```
Comment (thelonelyvulpes) [inline Neo4j.Driver/Neo4j.Driver/Experimental/FluentQueries/ExecutableQuery.cs]: We should remove this API the behaviour differs from WithParameters which overwrites previous with parameter calls,

Comment (RichardIrons-neo4j) [reply Neo4j.Driver/Neo4j.Driver/Experimental/FluentQueries/ExecutableQuery.cs]: Done this.

File: Neo4j.Driver/Neo4j.Driver/IResultCursor.cs [hunk 2]
```diff
@@ -60,14 +58,6 @@ public interface IResultCursor
         /// <returns>A task returning a <see cref="bool"/>. Task's result is true if there is any result to be consumed, false otherwise.</returns>
         Task<bool> FetchAsync();
 
-        /// <summary>
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/IResultCursor.cs]: Should this be removed form the interface? Is that not a breaking change to any customers that currently use it?

**Review by AndyHeap-NeoTech — APPROVED** [review]
Looks very good. I had one question about the result cursor interface, please see the comment.

File: Neo4j.Driver/Neo4j.Driver/IResultCursor.cs (hunk 2 above)
Comment (thelonelyvulpes) [reply Neo4j.Driver/Neo4j.Driver/IResultCursor.cs]: The Current still exists but it is coming from the IAsyncEnumerator<IRecord>. I think we might want a new definition though as the documentation will be wrong for how fetch works?

**Review by AndyHeap-NeoTech — APPROVED** [review]
Latest change looks to address the issues discussed.

File: Neo4j.Driver/Neo4j.Driver/Internal/Result/ResultCursor.cs [hunk 3]
```diff
@@ -136,5 +139,17 @@ public void Cancel()
         {
             _resultStream.Cancel();
         }
+
+        public IAsyncEnumerator<IRecord> GetAsyncEnumerator(
+            CancellationToken cancellationToken = new())
```
Comment (thelonelyvulpes) [inline Neo4j.Driver/Neo4j.Driver/Internal/Result/ResultCursor.cs]: public IAsyncEnumerator<IRecord> GetAsyncEnumerator(CancellationToken cancellationToken = default)
or CancellationToken.None

File: Neo4j.Driver/Neo4j.Driver/Experimental/QueryConfig.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        ImpersonatedUser = impersonatedUser;
+        BookmarkManager = bookmarkManager;
+        EnableBookmarkManager = enableBookmarkManager;
+    }
+}
+
+/// <summary>
+/// Configuration for running queries using the simplified API.
+/// </summary>
+public class QueryConfig<T> : QueryConfig
```
Comment (thelonelyvulpes) [inline Neo4j.Driver/Neo4j.Driver/Experimental/QueryConfig.cs:78]: This class is no longer used is it?

File: Neo4j.Driver/Neo4j.Driver/Internal/Result/ConsumableResultCursor.cs [hunk 5]
```diff
@@ -75,5 +82,15 @@ private void AssertNotConsumed()
                 throw ErrorExtensions.NewResultConsumedException();
             }
         }
+
+        public IAsyncEnumerator<IRecord> GetAsyncEnumerator(CancellationToken cancellationToken = new CancellationToken())
```
Comment (thelonelyvulpes) [inline Neo4j.Driver/Neo4j.Driver/Internal/Result/ConsumableResultCursor.cs]:
```suggestion
        public IAsyncEnumerator<IRecord> GetAsyncEnumerator(CancellationToken cancellationToken = CancellationToken.None)
```

**Pull request merged**